salt/minion.py

```diff
@@ -2926,7 +2930,38 @@ def nodegroup_match(self, tgt, nodegroups):
         return False
 
 
-class ProxyMinion(MinionManager):
+class ProxyMinionManager(MinionManager):
+    '''
+    Create the multi-minion interface but for proxy minions
+    '''
+
+    def _spawn_minions(self):
+        '''
+        Spawn all the coroutines which will sign in to masters
+        NOTE MinionManager._spawn_minions should be examined if any changes are made to
```

Comment:
Just the creation of the Minion or ProxyMinion object when iterating through the masters is different.  I thought about changing the function to take something like 'proxy_minion=True' and using that for a conditional that would create the correct type of object, but it seemed like a bolt-on.  Happy to refactor to make it better.  Maybe instead create another method called 'new_minion_object' that only creates the new object and override that method in ProxyMinionManager?